modules/tensor_mechanics/src/userobjects/ElementPropertyReadFile.C

```diff
@@ -110,6 +118,24 @@ ElementPropertyReadFile::readGrainData()
   initGrainCenterPoints();
 }
 
+void
+ElementPropertyReadFile::readBlockData()
+{
+  mooseAssert(_nblock > 0, "Error ElementPropertyReadFile: Provide non-zero number of blocks");
+  _data.resize(_nprop * _nblock);
+
+  MooseUtils::checkFileReadable(_prop_file_name);
+  std::ifstream file_prop;
+  file_prop.open(_prop_file_name.c_str());
+
+  for (unsigned int i = 0; i < _nblock; i++)
+    for (unsigned int j = 0; j < _nprop; j++)
+      if (!(file_prop >> _data[i * _nprop + j]))
+        mooseError("Error ElementPropertyReadFile: Premature end of file");
+
+  file_prop.close();
```

Comment:
This is almost the same as `readGrainData` ?! Surely this can be refactored. Also, using the `DelimitedFileReader` would be nice here.

Reply:
Yeah I will what I can do to refactor these. Thanks! I will checkout that file reader